calendar_resource/models/resource_calendar.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Laslabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime, time, timedelta
+
+from odoo import api, models
+
+
+class ResourceCalendar(models.Model):
+
+    _inherit = 'resource.calendar'
+
+    @api.model
+    def _get_conflicting_unavailable_intervals(self, intervals,
+                                               start_datetime, end_datetime):
+        """ Finds all unavailable datetime gaps in intervals argument that
+            overlap start_datetime and end_datetime args.
+
+        Args:
+            intervals (list): List of tuples containing the AVAILABLE working
+                datetimes between the start_datetime and end_datetime values.
+                Each tuple contains a start and stop datetime value.
+
+                The working datetime tuples to be supplied to this method can
+                be retrieved like so. Refer to its use in the
+                _check_resource_ids_working_times method in calendar.event in
+                this module.
+
+                    .. code-block:: python
+
+                    intervals = \
+                        resource.calendar_id.get_working_intervals_of_day(
+                            start_dt=datetime_start,
+                            end_dt=datetime_end,
+                            resource_id=resource.id,
+                        )
+
+            start_datetime, end_datetime (datetime): Datetime values
+                of which any unavailable intervals will be checked against
+                for overlaps.
+
+        Returns:
+            list: List containing any intervals of which the start
+            and end datetimes of those intervals overlap the
+            start_datetime and end_datetime args.
+
+        """
+        unavailable_intervals = self._get_unavailable_intervals(
+            intervals=intervals,
+            start_date=start_datetime.date(),
+            end_date=end_datetime.date(),
+        )
+
+        conflicting_intervals = []
+        for interval in unavailable_intervals:
+
+            if all((interval[0] < end_datetime,
+                    interval[1] > start_datetime)):
+                conflicting_intervals.append(interval)
+
+        return conflicting_intervals
+
+    @api.model
+    def _get_unavailable_intervals(self, intervals, start_date, end_date):
+        """ Finds any gaps between intervals, the beginning of start_date,
```

Comment:
hmm. I think in the docstring I'll put an example what the interval arg would look like. I'm using a method in the `resource` module that returns all the working times for the day in the following format:
`[(start_datetime, stop_datetime), (start_datetime, stop_datetime)]`. So basically a list of all the intervals available.